pennylane/workflow/_capture_qnode.py

```diff
@@ -200,9 +200,26 @@ def _(*args, qnode, shots, device, execution_config, qfunc_jaxpr, n_consts, batc
     consts = args[:n_consts]
     non_const_args = args[n_consts:]
 
-    partial_eval = partial(
-        device.eval_jaxpr, qfunc_jaxpr, consts, execution_config=execution_config
-    )
+    device_program, config = device.preprocess(execution_config)
```

Comment:
```suggestion
    device_program = device.preprocess_transforms(execution_config)
```
We setup the execution config before binding, so we don't actually need to do that step again. So we can use the more specific `preprocess_transforms`.

Reply:
Actually, idk if we can do that though ... 

```python
    device_program, config = device.preprocess(execution_config)
    ...
    partial_eval = partial(device.eval_jaxpr, qfunc_jaxpr, consts, execution_config=config)
    ...
    return jax.vmap(partial_eval, batch_dims[n_consts:])(*non_const_args)
```
We need the processed config for the `partial_eval`?

Reply:
Nvm, it is setup before bind is called, done in https://github.com/PennyLaneAI/pennylane/pull/7057/commits/